File: hackerrank/problem solving/algorithms/1 - warmup/9 - birthday cake candles/with morfo library/birthday_cake_candles.py
# https://www.hackerrank.com/challenges/birthday-cake-candles/problem?isFullScreen=true

import morfo


def main():
    morfo.skip_input_ln()
    candles = morfo.readln(int, list)
    print(birthday_cake_candles(candles))


# n: length of candles
# T: O(2n) => O(n)
# S: O(1) extra space
def birthday_cake_candles(candles):
    return candles.count(max(candles))


if __name__ == '__main__':
    main()

File: hackerrank/problem solving/algorithms/1 - warmup/9 - birthday cake candles/with morfo library/birthday_cake_candles.cs
// https://www.hackerrank.com/challenges/birthday-cake-candles/problem?isFullScreen=true

using Morfo.IO;

Reader.SkipInputLn();
var candles = Reader.ReadLn<int, int[]>();
Console.WriteLine(BirthdayCakeCandles(candles));

// n: length of candles
// T: O(2n) => O(n)
// S: O(1) extra space
int BirthdayCakeCandles(int[] candles)
{
    int max = candles.Max();
    return candles.Count(x => x == max);
}
